app/Domain/Contacts/Repositories/EloquentContactRepository.php

```diff
@@ -13,14 +13,19 @@ final class EloquentContactRepository implements ContactRepository
 {
     public function save(Contact $contact): void
     {
-        ContactModel::query()->create([
-            'id' => $contact->id->toString(),
-            'integration_id' => $contact->integrationId->toString(),
-            'type' => $contact->type->value,
-            'first_name' => $contact->firstName,
-            'last_name' => $contact->lastName,
-            'email' => $contact->email,
-        ]);
+        ContactModel::query()->updateOrCreate(
+            [
+                'id' => $contact->id->toString(),
+            ],
+            [
+                'id' => $contact->id->toString(),
```

Comment:
Is this still needed here? Because a new contact won't have an id, right?

Reply:
I think it is needed because a new contact has an `id` which is part of the domain object.
This `id` is a UUID and is created and set before saving the contact to the database.

Reply:
If the contact already has a UUID before it's saved to the DB, you need an `updateOrCreate()` an explicitly insert/update the `id` column yes. Another approach would be to not give new contacts a UUID in the domain model and check here if a create or update needs to be done based on the UUID being null. But `null` values are always annoying and the app should be able to generate its own UUIDs without issues instead of needing to rely on the ORM to do that, so I also slightly prefer this approach.

Reply:
Oh yeah, didn't fully realize it's a UUID. Carry on.